### ISSUE TITLE
Move test out of app/ but hardcoded path

### ISSUE THREAD

**Issue by luckydonald** [kenjis/ci-phpunit-test#211]
The current Structure:
`project/dist/app/tests`
But we want to sepereate it from the code we upload to the server:
`project/tests`.

Related to #171.

# Problem
Changing the settable pathes in `project/tests/Bootstrap.php` is not enough
```php
define('TESTPATH', __dir__.DIRECTORY_SEPARATOR);
$application_folder = TESTPATH.'../dist/app';
$system_path =        TESTPATH.'../dist/sys';
```
The place of the test folder are hardcoded to be in `APPPATH`.
[`CIPHPUnitTest.php:38`](https://github.com/kenjis/ci-phpunit-test/blob/1409b83cdf236bd79c812c41592c8e51b886754a/application/tests/_ci_phpunit_test/CIPHPUnitTest.php#L38)

![grafik](https://user-images.githubusercontent.com/2737108/31549686-382b0808-b02f-11e7-92c3-d8e1bf9b9a0a.png)

# Solution
With the define in the `Bootstrap.php` as above,
```php
define('TESTPATH', __dir__.DIRECTORY_SEPARATOR);  // or your path to the test folder
```

Replacing them via Regex seem to work:
>```regex
>APPPATH(\s*\.\s*['"])/?tests/
>```
>```php
>TESTPATH$1
>```

# TL;DR
1. Define `TESTPATH` in Bootstrap.php
2. replace `APPPATH . "tests/foo"` with `TESTPATH . "foo"`.
3. Praise Celestia for the sunshine.

**Comment by luckydonald** [comment]
Should I make a PR?

**Comment by kenjis** [comment]
> Should I make a PR?

Please do.

**Comment by luckydonald** [comment]
Perfect. Will do later today.

**Comment by kenjis** [comment]
@luckydonald Thank you for your PR!

**Comment by luckydonald** [comment]
@kenjis I realized, there is some kind of install procedure, 
The `README.md`, in the [**Upgrading via Composer**](https://github.com/kenjis/ci-phpunit-test/tree/34eb4c6cdc02975b56aa860500d39078e5ddd0be#upgrading-via-composer) section, talks about a  
```bash
update.php [-a <application_dir> -p <public_dir>]
```
moving composer files to the right folders.

Do you think we need to adjust something there if we still want a separate folder for tests?

I.e. `APPPATH = 'project_root/dist/application'` and `TESTPATH = 'project_root/tests'`:
```
project_root/
├── dist/
│   ├── application/
│   │   ├── controllers/
│   │   └── libraries/
│   │   ⠇
│   └── system/
├── tests/
│   ├── ci_phpunit_test/  ... don't touch! files ci-phpunit-test uses
│   ├── Bootstrap.php     ... bootstrap file for PHPUnit
│   ├── DbTestCase.php    ... DbTestCase class
│   ├── TestCase.php      ... TestCase class
│   ├── controllers/      ... put your controller tests
│   ├── libraries/        ... put your library tests
│   ├── mocks/
│   │   └── libraries/    ... mock libraries
│   ├── models/           ... put your model tests
│   └── phpunit.xml       ... config file for PHPUnit
└── vendor/

**Comment by luckydonald** [comment]
It seems to need an update, yes.
